hogvm/python/stl.py

```diff
@@ -0,0 +1,118 @@
+import time
+from typing import Any, Optional
+from collections.abc import Callable
+import re
+import requests
+
+from posthog.hogql.query import execute_hogql_query
+from posthog.models import Team
+
+
+def concat(name: str, args: list[Any], team: Optional[Team], stdout: Optional[list[str]], timeout: int):
+    def _to_concat_arg(arg) -> str:
+        if arg is None:
+            return ""
+        if arg is True:
+            return "true"
+        if arg is False:
+            return "false"
+        return str(arg)
+
+    return "".join([_to_concat_arg(arg) for arg in args])
+
+
+def match(name: str, args: list[Any], team: Optional[Team], stdout: Optional[list[str]], timeout: int):
+    return bool(re.search(re.compile(args[1]), args[0]))
+
+
+def toString(name: str, args: list[Any], team: Optional[Team], stdout: Optional[list[str]], timeout: int):
+    if args[0] is True:
+        return "true"
+    elif args[0] is False:
+        return "false"
+    elif args[0] is None:
+        return "null"
+    else:
+        return str(args[0])
+
+
+def toInt(name: str, args: list[Any], team: Optional[Team], stdout: Optional[list[str]], timeout: int):
+    try:
+        return int(args[0]) if name == "toInt" else float(args[0])
+    except ValueError:
+        return None
+
+
+def ifNull(name: str, args: list[Any], team: Optional[Team], stdout: Optional[list[str]], timeout: int):
+    if args[0] is not None:
+        return args[0]
+    else:
+        return args[1]
+
+
+def length(name: str, args: list[Any], team: Optional[Team], stdout: Optional[list[str]], timeout: int):
+    return len(args[0])
+
+
+def empty(name: str, args: list[Any], team: Optional[Team], stdout: Optional[list[str]], timeout: int):
+    return not bool(args[0])
+
+
+def notEmpty(name: str, args: list[Any], team: Optional[Team], stdout: Optional[list[str]], timeout: int):
+    return bool(args[0])
+
+
+def lower(name: str, args: list[Any], team: Optional[Team], stdout: Optional[list[str]], timeout: int):
+    return args[0].lower()
+
+
+def upper(name: str, args: list[Any], team: Optional[Team], stdout: Optional[list[str]], timeout: int):
+    return args[0].upper()
+
+
+def reverse(name: str, args: list[Any], team: Optional[Team], stdout: Optional[list[str]], timeout: int):
+    return args[0][::-1]
+
+
+def httpGet(name: str, args: list[Any], team: Optional[Team], stdout: Optional[list[str]], timeout: int):
```

Comment:
tbh this is the only bit im not super in favour of having _already_ as its the main place where we will have "dangerzone" thoughts in terms of long term running operations, DNS spoofing etc. If there's no rush, could we not add this later properly instead?

Reply:
(╯°□°)╯︵ ┻━┻

It's gone